ops/services/alerts/app_service_metrics/main.tf

```diff
@@ -319,3 +319,113 @@ customEvents
     threshold = 0
   }
 }
+
+resource "azurerm_monitor_scheduled_query_rules_alert" "fhir_batched_uploader_single_failure_detected" {
+  name                = "${var.env}-fhir-batched-uploader-single-failure-detected"
+  description         = "FHIRTestEventReporter failed to successfully complete"
+  location            = data.azurerm_resource_group.app.location
+  resource_group_name = var.rg_name
+  severity            = var.severity
+  frequency           = 5
+  time_window         = 11
+  enabled             = contains(var.disabled_alerts, "fhir_batched_uploader_single_failure_detected") ? false : true
+
+  data_source_id = var.app_insights_id
+
+  query = <<-QUERY
+requests
+${local.skip_on_weekends}
+| where timestamp >= ago(11m) 
+    and operation_Name =~ 'FHIRTestEventReporter' 
+    and success != true
+  QUERY
+
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 4
+  }
+
+  action {
+    action_group = var.action_group_ids
+  }
+}
+
+resource "azurerm_monitor_scheduled_query_rules_alert" "fhir_batched_uploader_function_not_triggering" {
+  name                = "${var.env}-fhir-batched-uploader-function-not-triggering"
+  description         = "FHIRTestEventReporter is not triggering on schedule"
+  location            = data.azurerm_resource_group.app.location
+  resource_group_name = var.rg_name
+  severity            = var.severity
+  frequency           = 5
+  time_window         = 7
+  enabled             = contains(var.disabled_alerts, "fhir_batched_uploader_function_not_triggering") ? false : true
+
+  data_source_id = var.app_insights_id
+
+  query = <<-QUERY
+requests
+${local.skip_on_weekends}
+| where timestamp >= ago(7m) 
+    and operation_Name =~ 'FHIRTestEventReporter'
+  QUERY
+
+  trigger {
+    operator  = "Equal"
+    threshold = 0
+  }
+
+  action {
+    action_group = var.action_group_ids
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "function_app_memory_alert" {
```

Comment:
Nice work on the function mem alert. I love that this alerting gap is being closed. Where did you get your threshold number? Have we had issues with our function memory before?

Reply:
We have not had any issues with the function memory, but now this function app is going to actually be running two batch uploads, CSV to RS's legacy pipeline and FHIR to RS's Universal Pipeline. I believe the 1200mb number came from the limit of app functions using 1.5 gb total, correct me if I'm wrong @johanna-skylight. 

Reviewing the function app right now in production it's always under 500 mb and actually closer to 200mb.

Reply:
That is correct the prod average seems to be 200mb so we would like to be alerted if we are getting closer to the 1.5GB limit that an app function has because that would mean we need to move the new function to be its own app. 

If you have a better way to define thresholds that is also appreciate it! :D

Reply:
Thank you for the info! :bee: The thought process makes total sense. With such a high threshold compared to typical usage, we'll know we have a problem if we hit it.

Nit: Could create a variable in the `_var.tf` file with 1200mb as the default. This would accomplish what we have here and allow future devs to set it at the module level per environment. That said, future devs could do this themselves if they need to, so feel free to skip it for now if we're ok with 1200mb being the value in all environments.

Thanks so much for answering my question!